torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -383,9 +384,68 @@ std::string dump_nccl_trace(
 control_plane::RegisterHandler dumpHandler{
     "dump_nccl_trace_pickle",
     [](const control_plane::Request& req, control_plane::Response& res) {
-      // TODO: c-p-i-o: params from the request need to go to dump_nccl_trace.
+      auto params = req.params();
+      auto includeCollectives = true;
+      auto includeStackTraces = true;
+      auto onlyActive = false;
+      size_t validParamCount = 0;
+
+      const auto& includeCollectivesIt = params.find("includecollectives");
+      if (includeCollectivesIt != params.end()) {
```

Comment:
wonder if instead of copypasta for parsing each one, you could have a loop over some tuples of (name, var, defaultval) like 
("includecollectives", includeCollectives, True) so that you only need to do the logic once?